### ISSUE TITLE
Tests are flaky

### ISSUE THREAD

**Issue by leonerd** [matrix-org/matrix-appservice-irc#166]
The tests that run under travis appear to be flaky. Commits can fail that subsequently pass after forcing a retest by pushing an empty commit.

An issue I have observed is the test:

```
  1) Matrix-to-IRC message bridging should bridge matrix topics as IRC topics
   Message:
     timeout: timed out after 5000 msec waiting for spec to complete
   Stacktrace:
     undefined
```

But I believe @lukebarnard1 has some others too.

**Comment by leonerd** [comment]
The detailed log from my failing test:

```
Matrix-to-IRC message bridging : should bridge matrix topics as IRC topics
==========================================================================
Add listener(irc.example_M-flibble) for fn=connect
Add listener(irc.example_M-flibble) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=connect
Add listener(irc.example_ro_bot_nick) for fn=join
2016-09-12 13:29:36 ERROR:BridgedClient <ro_bot_nick@irc.example#hbxhxrimee0cs> Timed out trying to join #coffee - trying again.
2016-09-12 13:29:36 DEBUG:BridgedClient <ro_bot_nick@irc.example#hbxhxrimee0cs> Joining channel #coffee
2016-09-12 13:29:36 ERROR:BridgedClient <ro_bot_nick@irc.example#vfspbq83fvkg> Timed out trying to join #coffee - trying again.
2016-09-12 13:29:36 DEBUG:BridgedClient <ro_bot_nick@irc.example#vfspbq83fvkg> Joining channel #coffee
2016-09-12 13:29:36 ERROR:BridgedClient <ro_bot_nick@irc.example#5g1owfudkzk0w> Timed out trying to join #coffee - trying again.
2016-09-12 13:29:36 DEBUG:BridgedClient <ro_bot_nick@irc.example#5g1owfudkzk0w> Joining channel #coffee
2016-09-12 13:29:36 ERROR:BridgedClient <ro_bot_nick@irc.example#igmgjfjbhw8c8> Timed out trying to join #coffee - trying again.
...
```

**Comment by lukebarnard1** [comment]
And on another unrelated test:

```
IRC-to-Matrix mode bridging : should set join_rules to 'invite' on +i.
======================================================================
Add listener(irc.example_ro_bot_nick) for fn=join
Add listener(irc.example_ro_bot_nick) for fn=connect
2016-09-12 12:46:16 DEBUG:BridgedClient <ro_bot_nick@irc.example#essvrrfucfswc> Joining channel #coffee
2016-09-12 12:46:16 ERROR:BridgedClient <ro_bot_nick@irc.example#hfefaq7e96o08> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:16 DEBUG:BridgedClient <ro_bot_nick@irc.example#hfefaq7e96o08> Joining channel #coffee
2016-09-12 12:46:16 ERROR:BridgedClient <ro_bot_nick@irc.example#ffmixpradn488> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:16 DEBUG:BridgedClient <ro_bot_nick@irc.example#ffmixpradn488> Joining channel #coffee
2016-09-12 12:46:16 ERROR:BridgedClient <ro_bot_nick@irc.example#8yrstpjk2ssow> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:16 DEBUG:BridgedClient <ro_bot_nick@irc.example#8yrstpjk2ssow> Joining channel #coffee
2016-09-12 12:46:17 ERROR:BridgedClient <ro_bot_nick@irc.example#jgvp191ol3sc4> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:17 DEBUG:BridgedClient <ro_bot_nick@irc.example#jgvp191ol3sc4> Joining channel #coffee
2016-09-12 12:46:18 ERROR:BridgedClient <ro_bot_nick@irc.example#50ibg6tdqe4gg> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:18 DEBUG:BridgedClient <ro_bot_nick@irc.example#50ibg6tdqe4gg> Joining channel #coffee
2016-09-12 12:46:18 ERROR:BridgedClient <ro_bot_nick@irc.example#bvmo16cza0gss> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:18 DEBUG:BridgedClient <ro_bot_nick@irc.example#bvmo16cza0gss> Joining channel #coffee
2016-09-12 12:46:19 ERROR:BridgedClient <ro_bot_nick@irc.example#kw8qqxk87b40s> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:19 DEBUG:BridgedClient <ro_bot_nick@irc.example#kw8qqxk87b40s> Joining channel #coffee
2016-09-12 12:46:19 ERROR:BridgedClient <ro_bot_nick@irc.example#9bygwlvx2nksk> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:19 DEBUG:BridgedClient <ro_bot_nick@irc.example#9bygwlvx2nksk> Joining channel #coffee
2016-09-12 12:46:20 ERROR:BridgedClient <ro_bot_nick@irc.example#kkdw773d3rkck> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:20 DEBUG:BridgedClient <ro_bot_nick@irc.example#kkdw773d3rkck> Joining channel #coffee
2016-09-12 12:46:20 ERROR:BridgedClient <ro_bot_nick@irc.example#1kgas6lkmuf4s> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:20 DEBUG:BridgedClient <ro_bot_nick@irc.example#1kgas6lkmuf4s> Joining channel #coffee
2016-09-12 12:46:21 ERROR:BridgedClient <ro_bot_nick@irc.example#114ihh3q2g5cw> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:21 DEBUG:BridgedClient <ro_bot_nick@irc.example#114ihh3q2g5cw> Joining channel #coffee
2016-09-12 12:46:21 ERROR:BridgedClient <ro_bot_nick@irc.example#c6vs3pdxqq888> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:21 DEBUG:BridgedClient <ro_bot_nick@irc.example#c6vs3pdxqq888> Joining channel #coffee
2016-09-12 12:46:21 ERROR:BridgedClient <ro_bot_nick@irc.example#jl0ryposlag48> Timed out trying to join #coffee - trying again.
2016-09-12 12:46:21 DEBUG:BridgedClient <ro_bot_nick@irc.example#jl0ryposlag48> Joining channel #coffee```

This appears to block the test for 5000ms, making it fail.

**Comment by Kegsay** [comment]
I have reason to believe this is down to the database which is `rm -rf`d between each test. https://github.com/matrix-org/matrix-appservice-irc/commit/41fccaa757789df47cb5a16b7a470e6232ed6c93

**Comment by lukebarnard1** [comment]
Tests are apparently still flaky, trying to fix in #247

**Comment by Kegsay** [comment]
@lukebarnard1 fixed it! \o/
